### PR TITLE
[HOT-FIX] SMIME signature status not displayed in email 

### PR DESCRIPTION
**Pull request by dab246**
## Issue

- https://github.com/linagora/tmail-flutter/pull/3061#issuecomment-2311540384
- SMIME signature status not displayed in email 
- Env: v0.13.0

## Root causes

- The value of the `X-SMIME-Status` header contains spaces

Screenshot 2024-08-27 at 14 21 27

- The `PresentationEmail` object used in `InformationSenderAndReceiverBuilder` has a null `emailHeader` property.

## Solution

- Use `header:X-SMIME-Status:asText` header parsed in email

## Dependent

- Need merged: https://github.com/linagora/jmap-dart-client/pull/94

## Resolved

- Web:

https://github.com/user-attachments/assets/1c0d8cdc-3c23-426a-82b9-2709a48fbe12

- Mobile (`Online/Offline`):

[demo-android.webm](https://github.com/user-attachments/assets/b652981b-7399-4076-b462-b8aea128c1a4)

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]
lgtm

**Comment by chibenwa** [comment]
Exactly which email/get request was done for this?

**Comment by dab246** [comment]
> Exactly which email/get request was done for this?

When opening an email, we call `Email/get` to get the email content details. Currently we only get the `headers` property of the email to check. This leads to this problem. We are proposing to get the value directly from the parsed header (`header:X-SMIME-Status:asText` ) to get its value.

- Current jmap request `Email/get`

```json
{
  "using": [
    "urn:ietf:params:jmap:core",
    "urn:ietf:params:jmap:mail",
    "urn:apache:james:params:jmap:mail:shares"
  ],
  "methodCalls": [
    [
      "Email/get",
      {
        "accountId": "694c6e72a66a8406190dac6e963019569cfee60931c113bf2e1515583a8cc988",
        "ids": [
          "bd3b9600-6443-11ef-8bca-6d53426328b6"
        ],
        "properties": [
          "bodyValues",
          "htmlBody",
          "attachments",
          "headers",
          "keywords",
          "mailboxIds",
          "messageId",
          "references"
        ],
        "fetchHTMLBodyValues": true
      },
      "c0"
    ]
  ]
}
```

File: contact/pubspec.lock [hunk 1]
```diff
@@ -611,8 +611,8 @@ packages:
     dependency: "direct main"
     description:
       path: "."
-      ref: main
-      resolved-ref: b1b0d407e5082f9d7afc88ccec8d05bf39ec992f
+      ref: add-smime-status-header-property-for-email
```
Comment (hoangdat) [inline contact/pubspec.lock:614]: please bump minor version for `jmap-dart-client` and update it here

File: test/features/email/presentation/extensions/email_extension_test.dart [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_header.dart';
+import 'package:jmap_dart_client/jmap/mail/email/individual_header_identifier.dart';
+import 'package:model/email/email_property.dart';
+import 'package:tmail_ui_user/features/email/presentation/extensions/email_extension.dart';
+import 'package:tmail_ui_user/features/email/presentation/model/smime_signature_status.dart';
+
+void main() {
+  group('EmailExtension::sMimeStatus::test', () {
+    test(
```
Comment (hoangdat) [inline test/features/email/presentation/extensions/email_extension_test.dart:11]: please add the cases: ` Good signature`, ` Bad signature`, ` Good Signatures`, .....

File: contact/pubspec.lock (hunk 1 above)
Comment (dab246) [reply contact/pubspec.lock:614]: Please review it at here https://github.com/linagora/jmap-dart-client/pull/95

File: test/features/email/presentation/extensions/email_extension_test.dart (hunk 2 above)
Comment (dab246) [reply test/features/email/presentation/extensions/email_extension_test.dart:11]: done

**Review by hoangdat — APPROVED** [review]

**Review by tddang-linagora — APPROVED** [review]

**Pull request merged**